src/{% if namespace_package %}{{namespace_package}}{% endif %}/{{ projectname.removeprefix(namespace_package) }}/__init__.py.jinja

```diff
@@ -1,12 +1,4 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) {{year}} {{orgname|capitalize}} contributors (https://github.com/{{orgname}})
 
-# flake8: noqa
-import importlib.metadata
-
-try:
-    __version__ = importlib.metadata.version(__package__ or __name__)
-except importlib.metadata.PackageNotFoundError:
-    __version__ = "0.0.0"
-
-del importlib
+from ._version import __version__, __version_tuple__
```

Comment:
Looks good, but how do we know it works? Can we test in the `copier_template_test`?

Reply:
It already checks for `__version__`. I added a check for `__version_tuple``. But I don't see how we could test that the value is correct.

Reply:
Um, you say in #59 "I don't think __version_tuple__ is standard. At least I couldn't find anything on it." This mean if it is changed, *all* our packages will break. Unless I misunderstood, I am not comfortable with that. What is wrong with using `importlib`?

Reply:
Nothing is wrong with `importlib`.
Neither `__version_tuple__` nor `__version__` are standard. The latter is basically ubiquitous at this point so people expect packages to have it. Whether we define it through `importlib`, a file written by `setuptools_scm`, or any other tool (e.g. `versioneer`) has no impact on the user.

The only question is whether we should have `__version_tuple__`. It is true that if we remove it, it would potentially break downstream code. But regardless of what tool we use, we can always construct the tuple by parsing `__version__`.
I think the tuple adds enough value to justify adding it because it allows users to compare versions more easily as shown in the issue: `if project.__version_tuple__ < (23, 8): ...`.

Reply:
But where are we defining `__version_tuple__`? I can't find the code writing `_version.py`. If it is just an implementation detail of `setuptools_scm` then we should not rely on it, since an update of that will break all our repos.

I am not suggesting to not define `__version_tuple__`. All I am saying is that we should not rely on some undocumented implementation detail that can break at any moment.

Reply:
It's the changes to pyproject.toml, in particular the `write_to` field. We also use this in Scipp now since we moved away from setup.cfg. So this is an actual setting of `setuptools_scm` and not an implementation detail.

Reply:
So why did you write "I don't think __version_tuple__ is standard. At least I couldn't find anything on it."?

Reply:
I was talking about the Python standard. `__version_tuple__` is not mentioned in the reference or any PEP. But neither is `__version__` (except for a comment on code style).
`setuptools_scm` does not explicitly document what it writes to the `_version.py` file either. But reading through their issues, it seems that these are not implementation details.

I did however find out that the `write_to` config option is deprecated. Replacing it in a separate PR.

Reply:
> But neither is `__version__` (except for a comment on code style).

I don't see how this is relevant. Defining `__version__` is great. But *defining it via an undocumented feature of a library*  is the problem.

> setuptools_scm does not explicitly document what it writes to the _version.py file either. But reading through their issues, it seems that these are not implementation details.

But why gamble (unless you find an explicit documentation)? Is using `importlib` so bad?

Reply:
> This seems unlikely to me as it would break a huge part of the ecosystem.

I would indeed be surprised if `__version__` would be removed, but as you said yourself, `__version_tuple__` is not something you can find much info on?

Reply:
Do we have any test that checks if this breaks? I believe the wheel builds are not running unit tests, so if `setuptools_scm` is updated we might not notice until after a release was made?